### DISCUSSION TITLE
Click event to show Pins giving me Uncaught TypeError: after adding Pin delete functionality

### DISCUSSION THREAD

**Discussion by BenJarrett** [nss-evening-cohort-14/evening-client-side#181] [Help Me!] [answered]
Ben Jarrett
I did. I read Juan's ticket #162. 
I tried to add the curly brackets to all of my exported functions, but it is giving me the same error on the start app when I finished updating the curly brackets. 

boardEvents.js?d674:8 Uncaught TypeError: Object(...) is not a function
    at HTMLHeadingElement.eval (boardEvents.js?d674:8) 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Not_a_function
https://www.educative.io/edpresso/what-is-an-uncaught-typeerror-in-javascript
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError

I have tried to re-write my displayPins function that happens on the button click I am listening for.
I have tried adding and then updating the curly brackets for the function on all files. 

https://github.com/BenJarrett/ASSIGNMENT-Firebase-Pinterest/blob/delete-pins/src/javascripts/Events/boardEvents.js

**Comment by drteresavasquez** [comment]
You're using query selector all. That works differently and needs a loop of some sort. Check the docs for how to add click events using that. 

That may be the issue here.

**Reply by BenJarrett** [reply]
Is there a way we did this in class recently?
I feel like I am forgetting a way that we did something similar to this on Almost Amazon

**Reply by drteresavasquez** [reply]
We did not use the query selector all. So there are no examples. We used unique IDs on everything.

**Reply by BenJarrett** [reply]
Okay, I see.

I want to do this the way we were taught in class. 
Would the class recording on Deleting and Filtering data be a good place to start if I wanted to re-write this using unique IDs?

**Reply by drteresavasquez** [reply]
I'm sure the videos cover it but it's also in most of the code that's in the DOM events file. We checked if the element included a string and then took the IDs off using the split method.

**Reply by BenJarrett** [reply]
Thanks Dr. T. 

I’m on my way back home now, so I’ll definitely go through this and try to figure it out tonight.

I’ll post the answer when I figure it all out!

**Reply by drteresavasquez** [reply]
Sounds like a plan. Try something and if it doesn't work, hit the ticket again and we can see if we can guide you. You should be able to employ the same techniques we did in class to get it solved.

**Comment by drteresavasquez** [comment]
Get pins in the file you shared needs curly brackets.

**Comment by drteresavasquez** [comment, marked as answer]
@BenJarrett following up on this ticket. Please update with a solution or status. Thanks!

**Reply by BenJarrett** [reply]
Hey! Sorry I didn't get back yet. 
I didn't notice that you followed up on this ticket.

This is a link to the current code of my branch:
https://github.com/BenJarrett/ASSIGNMENT-Firebase-Pinterest/tree/NEWEST-create-boards

First I added the curly brackets as you suggest and nothing changed. 

I decided to move on to create. 
I have that working as well. 
Only thing is, now when I hit my submit pin button, my click event to show pins, and on page load  I get this error 
GET https://pinterest-assignment-new-default-rtdb.firebaseio.com/pins.json?orderBy=%22boards_Id%22&equalTo=%22%22 400 (Bad Request)
And 
Uncaught (in promise) Error: Request failed with status code 400
    at createError (createError.js?2d83:16)
    at settle (settle.js?467f:17)
    at XMLHttpRequest.handleLoad (xhr.js?b50d:62)

So I know these are Firebase errors.
When I searched the first error in PostMan, I get "error": "Index not defined, add \".indexOn\": \"boards_Id\", for path \"/pins\", to the rules"
So I checked my firebase rules:
``` {
  "rules": {
    ".read": "now < 1617426000000",  // 2021-4-3
    ".write": "now < 1617426000000",  // 2021-4-3
      "boards": {
        ".indexOn": ["uid", "firebaseKey"]
      },
        "pins": {
        ".indexOn": ["uid", "firebaseKey", "boards_id"]
      },
  }
}
```

Not sure what exactly is wrong with my rules

**Reply by drteresavasquez** [reply]
Is this the same issue as the initial ticket?

**Reply by BenJarrett** [reply]
Not anymore.

But I was unable to answer the initial ticket.
I will create a new ticket for this!

**Reply by BenJarrett** [reply]
Created!
